models/source_list.py

```diff
@@ -31,6 +38,13 @@ class SourceList(Base, FileOnDiskMixin):
         )
```

Comment:
You didn't change this in this PR, but : do we need the is_sub column of the source list table?  Once the source_list points back to the image, we should be able to determine if it's a subtraction by using the is_sub property of the image object.  (If we wanted to do it as a database join, we'd join to image.ref_image is null and image.new_image is null.)

From the point of view of database normalization, I'd prefer not to have redundant (and, thus, possibly conflicting) information unless there's a compelling reason for it.

Reply:
I guess you are right. The idea was that it would be simple to tell the subtraction sources from the regular sources. Maybe there's an advantage when you can index on the `is_sub` column?

Reply:
If we index on the foreign key (which you should always do), and index in image.new_image and image.ref_image, then the database performance should be similar.

Reply:
got it.

Reply:
I can take out that column and add a property instead.